packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1624,17 +1626,19 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
     return RootRestorationScope(
       restorationId: widget.restorationScopeId,
       child: Shortcuts(
-        shortcuts: widget.shortcuts ?? WidgetsApp.defaultShortcuts,
         debugLabel: '<Default WidgetsApp Shortcuts>',
-        child: Actions(
-          actions: widget.actions ?? WidgetsApp.defaultActions,
-          child: FocusTraversalGroup(
-            policy: ReadingOrderTraversalPolicy(),
-            child: _MediaQueryFromWindow(
-              child: Localizations(
-                locale: appLocale,
-                delegates: _localizationsDelegates.toList(),
-                child: title,
+        shortcuts: widget.shortcuts ?? WidgetsApp.defaultShortcuts,
+        child: TextEditingShortcuts(
+          child: TextEditingActions(
+            additionalActions: widget.actions ?? WidgetsApp.defaultActions,
```

Comment:
Thoughts on the additionalActions approach here?

We need to think about what the user expects to happen when they pass in `shortcuts` and `actions`.  Previously these didn't override any text editing behavior because that was handled separately.

The two nested Shortcuts widgets here (one inside of TextEditingShortcuts) can't be combined because some text editing shortcuts bubble up to the upper widget when no EditableText is focused.

Reply:
I found the `additionalActions` strange (as commented on in the other file).

If you pass in shortcuts and actions here, I would expect them to override the defaults from TextEditingShortcuts and TextEditingActions. Could we just use a nested Action widget?

Reply:
I've updated this to follow the behavior below.  Let me know if it sounds good.

  * Passing `actions` gets rid of all default actions and all text editing actions.
    - Not just actions with conflicting keys.  So even if you just pass a Map with one action in it, NO default actions and NO text editing actions will be registered, just that one that was passed in.
    - This is how it worked before this PR too, just without any text editing actions.
 * Same with shortcuts.
 * additionalActions was removed in favor of nested Actions widgets.

Here is the commit with the relevant code: https://github.com/flutter/flutter/pull/75032/commits/5a7f658319e200702ca17d4c0d62c52b4a98e153